### PR TITLE
Fixes regression for not working embedded widgets

### LINKED ISSUE #8321: Embedding widgets with token_auth does not work anymore
Major regression in 2.14.0: embedding widgets is broken.

Reported in forum http://forum.piwik.org/read.php?2,127970 and reproduced on demo.

This is a bad regression, as many users will rely on this feature to embed and share their Piwik stats with others.

### PR DESCRIPTION
**Pull request by sgiehl**
fixes #8321

### CODE REVIEW

File: core/API/Request.php [hunk 1]
```diff
@@ -323,14 +323,14 @@ private static function forceReloadAuthUsingTokenAuth($tokenAuth)
 
     private static function shouldReloadAuthUsingTokenAuth($request)
     {
-        if (!isset($request['token_auth'])) {
```
Comment (sgiehl) [inline core/API/Request.php]: Problem here was, that ```$request``` is ```null```. ```Common::getRequestVar()``` has a fallback to use ```$POST + $GET``` in that case.

File: core/API/Request.php [hunk 2]
```diff
@@ -323,6 +323,10 @@ private static function forceReloadAuthUsingTokenAuth($tokenAuth)
 
     private static function shouldReloadAuthUsingTokenAuth($request)
     {
+        if (is_null($request)) {
```
Comment (sgiehl) [inline core/API/Request.php]: Problem is, that ```$request``` is ```null```. So the Auth was never reloaded.
I had searched for a better solution. But I didn't find another without changing the current logic.
There seems to be no way to determine whether a parameter is set in the request but empty. ```Common::getRequestVar()``` returns the default (or throws exception) if the parameter is missing or empty. Imho returning the empty value would be correct here, but I guess that might break other stuff in piwik to change that behavior.

**Comment by mattab** [comment]
>  Imho returning the empty value would be correct here, but I guess that might break other stuff in piwik to change that behavior.

If you think it would be correct behavior, maybe it's worth trying the correct fix and see if all our tests would still pass? it may be a bit risky but in general it's worthwile trying the "Correct fix approach" in case it may just work :-)

**Comment by sgiehl** [comment]
Maybe we should not try to do that for a minor fix version.... But I could
try to change that after 2.14.1 was released.

**Comment by mattab** [comment]
Makes sense to limit risk for 2.14.1 :+1:

**Comment by mattab** [comment]
looks good, thanks for looking into this regression @sgiehl 

Added tiny commit https://github.com/piwik/piwik/commit/1b30b7fbe9720b5de758b790d422ee57d1556716

**Pull request merged**